
From: Nick Piggin <nickpiggin@yahoo.com.au>

The attached patch is required to work correctly with the CPU hotplug
framework.  John Hawkes reports successful booting with this.


Index: linux-2.6.5/kernel/sched.c
===================================================================
--- linux-2.6.5.orig/kernel/sched.c	2004-04-16 15:06:46.000000000 +0200
+++ linux-2.6.5/kernel/sched.c	2004-04-16 15:06:46.000000000 +0200
@@ -3426,6 +3426,8 @@
 	runqueue_t *rq = cpu_rq(cpu);
 	int local = 1;
 
+	lock_cpu_hotplug();
+
 	spin_lock_irqsave(&rq->lock, flags);
 
 	if (cpu == smp_processor_id() || cpu_is_offline(cpu)) {
@@ -3444,6 +3446,8 @@
 		wake_up_process(rq->migration_thread);
 		wait_for_completion(&req.done);
 	}
+
+	unlock_cpu_hotplug();
 }
 
 #ifdef ARCH_HAS_SCHED_DOMAIN
